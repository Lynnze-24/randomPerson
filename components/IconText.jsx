import React from 'react'
import { View, Text,StyleSheet} from 'react-native'
import { MaterialCommunityIcons,Ionicons  } from '@expo/vector-icons';

import cusTheme from '../theme/Theme';
import useMediaQueries from '../hooks/MediaQueries';

const IconText = ({icon,text,type,style,isTable}) => {
    const {isLessThanLG,isLessThanXSM}= useMediaQueries()
    return (
        <View style={[isTable?styles.containerTable:styles.containerTile,style]}>
            {
                type==="Ionicons"?( <Ionicons testID='icon' name="location-outline" size={24} color={cusTheme.primary} />)
                                 :(<MaterialCommunityIcons testID='icon' name={icon} size={24} color={cusTheme.primary} />)
            }
            <Text style={isTable ? [styles.text,
                                    isLessThanLG && styles.textLG,
                                    isLessThanXSM && styles.textXSM]

                                  : styles.text
                                    }>{text}</Text>
        </View>
    )
}


const styles = StyleSheet.create({
    containerTable: {
        flex:1,
      flexDirection:'row',
      height:'100%',
      alignItems:'center',
   },
   containerTile: {
    flexDirection:'row',
    alignItems:'center',
 },
   text:{
       textAlign:'left',
       marginLeft:2,
       fontSize:13,
   },
   textLG:{
    width:100
   },
   textXSM:{
    width:'auto'
   }
})

export default IconText
