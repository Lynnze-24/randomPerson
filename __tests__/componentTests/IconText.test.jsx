import {render, fireEvent } from "@testing-library/react-native";
import IconText from "../../components/IconText";
import renderer from 'react-test-renderer';



it('matches snapShot', () => {
    const comp = renderer.create(<IconText icon="email-outline" text="Email" />).toJSON();
    expect(comp).toMatchSnapshot();
});

it('should render component correctly',()=>{ 
    
    const {getByText}=render(
                        <IconText icon="email-outline" text="Email" />
                        );
    //expo vector icons were ignored in transformpatterns so not show up here
        getByText("Email")
        
})
