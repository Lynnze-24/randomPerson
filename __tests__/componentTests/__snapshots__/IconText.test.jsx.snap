// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapShot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <Text />
  <Text
    style={
      Object {
        "fontSize": 13,
        "marginLeft": 2,
        "textAlign": "left",
      }
    }
  >
    Email
  </Text>
</View>
`;
